### PR TITLE
Negotiation subtask 1.1

### PR DESCRIPTION
**Pull request by simantvermasap**
Task 1: Check and fix schemas and examples in "negotiation"

### CODE REVIEW

File: negotiation/message/contract-agreement-message.json [hunk 1]
```diff
@@ -5,16 +5,15 @@
...
     "@type": "odrl:Agreement",
     "@id": "e8dc8655-44c2-46ef-b701-4cffdc2faa44",
+    "odrl:target": { "@id": "urn:uuid:3dd1add4-4d2d-569e-d634-8394a8836d23" },
     "dspace:timestamp": {
       "@value": "2023-01-01T01:00:00Z",
       "@type": "xsd:dateTime"
     },
     "dspace:consumerId": "...",
-    "dspace:providerId": "...",
-    "odrl:permission": {
```
Comment (sebbader-sap) [inline negotiation/message/contract-agreement-message.json:14]: Please add a permission. E.g. from https://github.com/International-Data-Spaces-Association/ids-specification/blob/9818c395cc73f827543d1290f09fa6a44b56dee7/negotiation/message/example/contract.agreement.message.http.transfer.json#L12

File: negotiation/message/contract-negotiation-termination-message.json [hunk 2]
```diff
@@ -2,7 +2,8 @@
   "@context":  "https://w3id.org/dspace/v0.8/context.json",
   "@type": "dspace:ContractNegotiationTerminationMessage",
   "dspace:processId": "urn:uuid:a343fcbf-99fc-4ce8-8e9b-148c97605aab",
-  "dspace:code": "...",
+  "dspace:code": {
```
Comment (sebbader-sap) [inline negotiation/message/contract-negotiation-termination-message.json]: Why is this a JSON Object now? I thought it should be a free-text field (=string)...

File: negotiation/message/contract-offer-message.json [hunk 3]
```diff
@@ -5,8 +5,7 @@
   "dspace:offer": {
     "@type": "odrl:Offer",
     "@id": "...",
-    "odrl:target": { "@id": "urn:uuid:3dd1add8-4d2d-569e-d634-8394a8836a88" },
-    ...
+    "odrl:target": { "@id": "urn:uuid:3dd1add8-4d2d-569e-d634-8394a8836a88" }
   },
```
Comment (sebbader-sap) [inline negotiation/message/contract-offer-message.json:20]: Same as above, please add a permission too.

File: negotiation/message/contract-request-message.json [hunk 4]
```diff
@@ -7,8 +7,7 @@
   "dspace:offer": {
     "@type": "odrl:Offer",
     "@id": "urn:uuid:2828282:3dd1add8-4d2d-569e-d634-8394a8836a89",
-    "odrl:target": { "@id": "urn:uuid:3dd1add8-4d2d-569e-d634-8394a8836a88"},
-    ...
+    "odrl:target": { "@id": "urn:uuid:3dd1add8-4d2d-569e-d634-8394a8836a88"}
```
Comment (sebbader-sap) [inline negotiation/message/contract-request-message.json]: again, please put a permission in.

File: negotiation/message/schema/contract-agreement-message-schema.json [hunk 5]
```diff
@@ -14,58 +14,56 @@
...
-					"required": true
+					"const": "dspace:ContractAgreementMessage"
 				},
 				"dspace:processId": {
-					"type": "string",
-					"required": true
+					"type": "string"
 				},
-				"odrl:agreement": {
+				"dspace:agreement": {
```
Comment (sebbader-sap) [inline negotiation/message/schema/contract-agreement-message-schema.json:26]: `agreement` is in the ODRL namespace, so please change back.

File: negotiation/message/schema/contract-agreement-verification-message-schema.json [hunk 6]
```diff
@@ -23,14 +23,18 @@
 				"dspace:processId": {
 					"type": "string"
 				},
-				"cred:subject": {
+				"cred:credentialSubject": {
```
Comment (sebbader-sap) [inline negotiation/message/schema/contract-agreement-verification-message-schema.json]: Please remove all clauses in the "cred" and "sec" namespace. That was an idea of me, which did not make it through the community review.

File: negotiation/message/schema/contract-negotiation-schema.json [hunk 7]
```diff
@@ -14,23 +14,29 @@
...
-					"required": true
+					"const": "https://w3id.org/dspace/v0.8/context.json"
 				},
 				"@type": {
 					"type": "string",
-					"const": "dspace:ContractNegotiation",
-					"required": true
+					"const": "dspace:ContractNegotiation"
 				},
 				"dspace:processId": {
```
Comment (sebbader-sap) [inline negotiation/message/schema/contract-negotiation-schema.json]: I think with the latest changes, `processId` was renamed to `processPid`. Can you please verify?

File: negotiation/message/schema/contract-offer-message-schema.json [hunk 8]
```diff
@@ -14,27 +14,36 @@
...
-					"required": true
+					"const": "dspace:ContractOfferMessage"
 				},
 				"dspace:processId": {
-					"type": "string",
-					"required": true
+					"type": "string"
 				},
-				"odrl:offer": {
+				"dspace:offer": {
```
Comment (sebbader-sap) [inline negotiation/message/schema/contract-offer-message-schema.json:26]: Same as before with `agreement`: `offer` shall stay in the ODRL namespace.

File: negotiation/message/schema/contract-request-message-schema.json [hunk 9]
```diff
@@ -14,27 +14,36 @@
...
-					"const": "dspace:ContractRequestMessage",
-					"required": true
+					"const": "dspace:ContractRequestMessage"
 				},
 				"dspace:processId": {
-					"type": "string",
-					"required": true
+					"type": "string"
 				},
-				"odrl:offer": {
```
Comment (sebbader-sap) [inline negotiation/message/schema/contract-request-message-schema.json:29]: same as above.

**Review by sebbader-sap — CHANGES_REQUESTED** [review]

File: negotiation/message/schema/contract-negotiation-schema.json (hunk 7 above)
Comment (simantvermasap) [reply negotiation/message/schema/contract-negotiation-schema.json]: Couldn't find a reference to processPid. Still showing as processId

File: negotiation/message/schema/contract-agreement-verification-message-schema.json (hunk 6 above)
Comment (simantvermasap) [reply negotiation/message/schema/contract-agreement-verification-message-schema.json]: does this mean keeping 'cred' and 'sec' blank arrays?

File: negotiation/message/contract-agreement-message.json (hunk 1 above)
Comment (simantvermasap) [reply negotiation/message/contract-agreement-message.json:14]: done

File: negotiation/message/contract-negotiation-termination-message.json (hunk 2 above)
Comment (simantvermasap) [reply negotiation/message/contract-negotiation-termination-message.json]: reverted back

File: negotiation/message/contract-offer-message.json (hunk 3 above)
Comment (simantvermasap) [reply negotiation/message/contract-offer-message.json:20]: done

File: negotiation/message/contract-request-message.json (hunk 4 above)
Comment (simantvermasap) [reply negotiation/message/contract-request-message.json]: done

File: negotiation/message/schema/contract-offer-message-schema.json (hunk 8 above)
Comment (simantvermasap) [reply negotiation/message/schema/contract-offer-message-schema.json:26]: done

File: negotiation/message/schema/contract-request-message-schema.json (hunk 9 above)
Comment (simantvermasap) [reply negotiation/message/schema/contract-request-message-schema.json:29]: done

File: negotiation/message/schema/contract-offer-message-schema.json (hunk 8 above)
Comment (sebbader-sap) [reply negotiation/message/schema/contract-offer-message-schema.json:26]: Sorry, my bad, there is no `odrl:offer`, so please change back to `dspace:offer`.

File: negotiation/message/contract-agreement-message.json (hunk 1 above)
Comment (sebbader-sap) [reply negotiation/message/contract-agreement-message.json:14]: The `odrl:leftOperand` in my linked example is not that good. Better use something like this:

```
    "odrl:permission": {
      "@id": "some-permission-id",
      "odrl:action": { "@id": "odrl:use" },
      "odrl:constraint": {
        "odrl:leftOperand": { "@id": "odrl:dateTime" },
        "odrl:operand": { "@id": "odrl:lteq" },
        "odrl:rightOperand": { "@value": "2023-12-31T06:00Z", "@type": "xsd:dateTime" }
      }
    }
```

File: negotiation/message/schema/contract-agreement-message-schema.json [hunk 10]
```diff
@@ -14,58 +14,56 @@
...
-					"required": true
+					"const": "dspace:ContractAgreementMessage"
 				},
 				"dspace:processId": {
-					"type": "string",
-					"required": true
+					"type": "string"
 				},
 				"odrl:agreement": {
 					"$ref": "https://w3id.org/dspace/schemas/v0.8/AgreementSchema#definitions/Agreement",
```
Comment (sebbader-sap) [inline negotiation/message/schema/contract-agreement-message-schema.json]: Please also correct the link to the contract.json schema file.

**Review by sebbader-sap — APPROVED** [review]

**Pull request merged**